diffkemp/diffkemp.py

```diff
@@ -1,143 +1,259 @@
-        ap.error("-f requires -p to be entered")
+    # "compare" sub-command
+    compare_ap = sub_ap.add_parser("compare",
+                                   help="compare generated snapshots for "
+                                        "semantic equality")
+    compare_ap.add_argument("snapshot_dir_old",
+                            help="directory with the old snapshot")
+    compare_ap.add_argument("snapshot_dir_new",
+                            help="directory with the new snapshot")
+    compare_ap.add_argument("--show-diff",
+                            help="show diff for non-equal functions",
+                            action="store_true")
+    compare_ap.add_argument("--regex-filter",
+                            help="filter function diffs by given regex")
+    compare_ap.add_argument("--report-stat",
+                            help="report statistics of the analysis",
+                            action="store_true")
+    compare_ap.add_argument("--kernel-dirs",
+                            nargs=2,
+                            help="specify root dirs for the compared kernels")
+    compare_ap.add_argument("--function", "-f",
+                            help="compare only selected function")
+    compare_ap.add_argument("--control-flow-only",
+                            help=SUPPRESS,
+                            action="store_true")
+    compare_ap.add_argument("--show-empty-diff",
+                            help=SUPPRESS,
+                            action="store_true")
+    compare_ap.add_argument("--semdiff-tool",
+                            help=SUPPRESS,
+                            choices=["llreve"])
+    compare_ap.set_defaults(func=compare)
+    return ap
 
 
 def run_from_cli():
-    """ Main method to run the tool. """
+    """Main method to run the tool."""
     ap = __make_argument_parser()
     args = ap.parse_args()
-    check_args(args, ap)
-
-    try:
-        first_builder = LlvmKernelBuilder(args.src_version, args.modules_dir,
-                                          args.debug, args.rebuild)
-        if args.module:
-            first_mods = {
-                args.module: first_builder.build_module(args.module)
-            }
-        elif args.all_modules:
-            first_mods = first_builder.build_all_modules()
-        elif args.file:
-            first_mods = {args.file: first_builder.build_file(args.file)}
-        else:
-            first_mods = first_builder.build_modules_with_params()
-
-        second_builder = LlvmKernelBuilder(args.dest_version, args.modules_dir,
-                                           args.debug, args.rebuild)
-        if args.module:
-            second_mods = {
-                args.module: second_builder.build_module(args.module)
-            }
-        elif args.all_modules:
-            second_mods = second_builder.build_all_modules()
-        elif args.file:
-            second_mods = {args.file: second_builder.build_file(args.file)}
+    args.func(args)
```

Comment:
This does not work when diffkemp is run without any arguments (the func attribute is not created in that case). Setting the required attribute on the object returned by `add_subparsers` should fix this issue and notify the user in case that they have to specify the subparser.